### PR TITLE
SIGINT-1980

### PR DESCRIPTION
**Pull request by jahid1209**
- Updated gradle version to 7.3.
- Updated license.
- Updated `SOAP` authentication to ensure compatibility with `java 17`.

### CODE REVIEW

File: src/main/java/com/synopsys/integration/coverity/ws/ClientAuthenticationHandlerWSS.java [hunk 1]
```diff
@@ -7,93 +7,86 @@
...
-        SOAPMessage msg = smc.getMessage();
-        if (outbound) {
-            try {
-                ProcessingContext context = xwssProcessor.createProcessingContext(msg);
-                context.setSOAPMessage(msg);
-                SOAPMessage secureMsg = xwssProcessor.secureOutboundMessage(context);
-                smc.setMessage(secureMsg);
-            } catch (Exception e) {
-                throw new RuntimeException(e);
+                SOAPElement usernameElement = usernameToken.addChildElement(USERNAME, WSS_AUTH_PREFIX);
```
Comment (lokeshbd) [inline src/main/java/com/synopsys/integration/coverity/ws/ClientAuthenticationHandlerWSS.java]: Does SOAP encodes the username and password internally? if not please add te encoding logic for both username and password.

Comment (jahid1209) [reply src/main/java/com/synopsys/integration/coverity/ws/ClientAuthenticationHandlerWSS.java]: As the server accepts plain text in the header, we cannot introduce any encoding, unfortunately!

File: src/main/java/com/synopsys/integration/coverity/ws/ClientAuthenticationHandlerWSS.java [hunk 2]
```diff
@@ -7,93 +7,86 @@
...
-import java.nio.charset.StandardCharsets;
-import java.util.HashSet;
-import java.util.Set;
-
-import javax.security.auth.callback.Callback;
-import javax.security.auth.callback.CallbackHandler;
-import javax.security.auth.callback.UnsupportedCallbackException;
 import javax.xml.namespace.QName;
-import javax.xml.soap.SOAPMessage;
+import javax.xml.soap.*;
```
Comment (sig-tithi) [inline src/main/java/com/synopsys/integration/coverity/ws/ClientAuthenticationHandlerWSS.java]: can we just import what is needed rather than *

**Review by sig-tithi — APPROVED** [review]

File: src/main/java/com/synopsys/integration/coverity/ws/ClientAuthenticationHandlerWSS.java (hunk 2 above)
Comment (jahid1209) [reply src/main/java/com/synopsys/integration/coverity/ws/ClientAuthenticationHandlerWSS.java]: Done!

**Pull request merged**